To: Linus Walleij <linus.walleij@xxxxxxxxxx>, Bartosz Golaszewski <brgl@xxxxxxxx>, Rob Herring <robh+dt@xxxxxxxxxx>, Krzysztof Kozlowski <krzysztof.kozlowski+dt@xxxxxxxxxx>, linux-gpio@xxxxxxxxxxxxxxx, devicetree@xxxxxxxxxxxxxxx, linux-kernel@xxxxxxxxxxxxxxx
Subject: [PATCH v2 2/2] gpio: pca953x: introduce support for nxp,pcal6408
From: Nate Drude <nate.d@xxxxxxxxxxxxx>
Date: Tue, 13 Sep 2022 13:24:36 -0500
Cc: eran.m@xxxxxxxxxxxxx, Nate Drude <nate.d@xxxxxxxxxxxxx>
References: <20220913182436.589713-1-nate.d@variscite.com>

The NXP PCAL6408 is the 8-bit version of PCAL6416.

Signed-off-by: Nate Drude <nate.d@xxxxxxxxxxxxx>

Source: https://www.spinics.net/lists/kernel/msg4513613.html
---
 drivers/gpio/gpio-pca953x.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/kernel/kernel-jammy-src/drivers/gpio/gpio-pca953x.c.orig	2025-10-10 18:33:18.382888723 -0700
+++ b/kernel/kernel-jammy-src/drivers/gpio/gpio-pca953x.c	2025-10-10 18:33:20.086958601 -0700
@@ -88,6 +88,7 @@
 	{ "pca9575", 16 | PCA957X_TYPE | PCA_INT, },
 	{ "pca9698", 40 | PCA953X_TYPE, },
 
+	{ "pcal6408", 16 | PCA953X_TYPE | PCA_LATCH_INT, },
 	{ "pcal6416", 16 | PCA953X_TYPE | PCA_LATCH_INT, },
 	{ "pcal6524", 24 | PCA953X_TYPE | PCA_LATCH_INT, },
 	{ "pcal9535", 16 | PCA953X_TYPE | PCA_LATCH_INT, },
@@ -1214,6 +1214,7 @@
 	{ .compatible = "nxp,pca9575", .data = OF_957X(16, PCA_INT), },
 	{ .compatible = "nxp,pca9698", .data = OF_953X(40, 0), },
 
+	{ .compatible = "nxp,pcal6408", .data = OF_953X(8, PCA_LATCH_INT), },
 	{ .compatible = "nxp,pcal6416", .data = OF_953X(16, PCA_LATCH_INT), },
 	{ .compatible = "nxp,pcal6524", .data = OF_953X(24, PCA_LATCH_INT), },
 	{ .compatible = "nxp,pcal9535", .data = OF_953X(16, PCA_LATCH_INT), },
